Mathlib/Tactic/Relation/Rfl.lean

```diff
@@ -38,22 +38,28 @@ initialize registerBuiltinAttribute {
     reflExt.add (decl, key) kind
 }
 
-open Elab.Tactic in
-/--
+/-- `MetaM` version of the `rfl` tactic.
+
 This tactic applies to a goal whose target has the form `x ~ x`, where `~` is a reflexive
 relation, that is, a relation which has a reflexive lemma tagged with the attribute [refl].
 -/
-elab_rules : tactic
-| `(tactic| rfl) => withMainContext do
-  let tgt ← getMainTarget
-  let .app (.app rel _) _ := tgt
-    | throwError "reflexivity lemmas only apply to binary relations, not {indentExpr tgt}"
+def _root_.Lean.MVarId.rfl (goal : MVarId) : MetaM (List MVarId) := do
```

Comment:
```suggestion
def _root_.Lean.MVarId.rfl (goal : MVarId) : MetaM Unit := do
```
I don't think `rfl` should ever leave subgoals, it is supposed to be a closing tactic.

Reply:
You can then use `liftMetaFinishingTactic` below.

Reply:
I agree, but having a `MetaM (List MVarId)` return type also lets is fit naturally into the notion of a "meta tactic" (so that we can just use e.g. `liftMetaTactic`, and some other functions which expect `MetaM (List MVarId)`. What's the optimal solution here? Standard funtion to turn a `MetaM Unit` return-type function into a `MetaM (List MVarId)` one?

Reply:
Oops, bitten by non-live review comments...